src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -385,27 +385,28 @@ public int CompareTo(object? obj)
         /// <returns>A value less than zero if this is less than <paramref name="other"/>, zero if this is equal to <paramref name="other"/>, or a value greater than zero if this is greater than <paramref name="other"/>.</returns>
         public int CompareTo(Half other)
         {
-            if (this < other)
-            {
-                return -1;
-            }
+            bool thisIsNaN = IsNaN(this);
```

Comment:
In general I think comparing against NaN is not a common operation, so I would expect that handling the `NaN` case in the end is the right way to go? But, if perf numbers suggest otherwise, I'd be good with this change.